src/three-domain-secure/component.jsx

```diff
@@ -115,47 +60,75 @@ const parseMerchantPayload = ({
 };
 
 export interface ThreeDomainSecureComponentInterface {
-  isEligible(): Promise<boolean>;
-  show(): ZoidComponent<void>;
+  isEligible(payload: MerchantPayloadData): Promise<boolean>;
+  show(): ZalgoPromise<threeDSResponse>;
 }
+
+type Update3DSTokenResponse = {|
+  updateTokenizedCreditCardWithExternalThreeDSecure: {|
+    paymentMethod: {|
+      id: string,
+    |},
+  |},
+|};
+
 export class ThreeDomainSecureComponent {
+  fastlaneNonce: string;
   logger: LoggerType;
-  request: Request;
+  restClient: RestClient;
+  graphQLClient: GraphQLClient;
   sdkConfig: SdkConfig;
   authenticationURL: string;
+  threeDSIframe: ZoidComponent<TDSProps>;
 
   constructor({
     logger,
-    request,
+    restClient,
+    graphQLClient,
     sdkConfig,
   }: {|
     logger: LoggerType,
-    request: Request,
+    restClient: RestClient,
+    graphQLClient: GraphQLClient,
     sdkConfig: SdkConfig,
   |}) {
     this.logger = logger;
-    this.request = request;
+    this.restClient = restClient;
+    this.graphQLClient = graphQLClient;
     this.sdkConfig = parseSdkConfig({ sdkConfig, logger });
   }
 
   async isEligible(merchantPayload: MerchantPayloadData): Promise<boolean> {
     const data = parseMerchantPayload({ merchantPayload });
+    this.fastlaneNonce = merchantPayload.nonce;
```

Comment:
There should be some validation on the nonce or the merchant payload. Ran into an issue where undefined was being passed to the GQL call. I believe this is going to be addressed by DTPPCPSDK-2861